### PR TITLE
Add specific events for interactions

### PR DESCRIPTION
**Pull request by KILLEliteMaste**
- ApplicationCommandCreateEvent
- MessageComponentCreateEvent
- ButtonClickEvent

### CODE REVIEW

File: javacord-api/src/main/java/org/javacord/api/event/interaction/ApplicationCommandCreateEvent.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+ * An application command create event.
+ */
+public interface ApplicationCommandCreateEvent extends Event {
+
+    /**
+     * Gets the created interaction.
+     *
+     * @return The interaction.
+     */
+    Interaction getInteraction();
```
Comment (coaxial1) [inline javacord-api/src/main/java/org/javacord/api/event/interaction/ApplicationCommandCreateEvent.java:18]: Why do we have this? Shouldn't it always be an ApplicationCommandInteraction in this event?
This sould be returning the substype imo, and the following optional method can be removed

File: javacord-api/src/main/java/org/javacord/api/event/interaction/ButtonClickEvent.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+ * A button click event.
+ */
+public interface ButtonClickEvent extends Event {
+
+    /**
+     * Gets the created interaction.
+     *
+     * @return The interaction.
+     */
+    Interaction getInteraction();
```
Comment (coaxial1) [inline javacord-api/src/main/java/org/javacord/api/event/interaction/ButtonClickEvent.java:19]: Same as in ApplicationCommandCreateEvent

File: javacord-api/src/main/java/org/javacord/api/event/interaction/MessageComponentCreateEvent.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+ * A message component interaction create event.
+ */
+public interface MessageComponentCreateEvent extends Event {
+
+    /**
+     * Gets the created interaction.
+     *
+     * @return The interaction.
+     */
+    Interaction getInteraction();
```
Comment (coaxial1) [inline javacord-api/src/main/java/org/javacord/api/event/interaction/MessageComponentCreateEvent.java:18]: Same

**Review by coaxial1 — CHANGES_REQUESTED** [review]
BTW, I couldn't read it from the code somehow, but can we already attach a button click listener to a ``Button`` (e.g. after we added one to a message we just sent)?

File: javacord-api/src/main/java/org/javacord/api/event/interaction/ApplicationCommandCreateEvent.java (hunk 1 above)
Comment (KILLEliteMaste) [reply javacord-api/src/main/java/org/javacord/api/event/interaction/ApplicationCommandCreateEvent.java:18]: This is exactly the interaction I mentioned in Discord. You also said this should not be removed as this is the top level interaction which someone might want

Comment (coaxial1) [reply javacord-api/src/main/java/org/javacord/api/event/interaction/ApplicationCommandCreateEvent.java:18]: I meant we shouldn't remove the top level interaction listener. On the top leven interaction create event, there should be a way to get the top level interaction. I thought you were talking about removing it from the top level create event in favor of the casting methods we added to it.

Comment (coaxial1) [reply javacord-api/src/main/java/org/javacord/api/event/interaction/ApplicationCommandCreateEvent.java:18]: Somehow I didn't realize that you're talking about the new events :D

**Comment by KILLEliteMaste** [comment]
> BTW, I couldn't read it from the code somehow, but can we already attach a button click listener to a `Button` (e.g. after we added one to a message we just sent)?
I haven't added this in the PR as I just wanted to have the more specific event. This could also be added in a followup PR

File: javacord-api/src/main/java/org/javacord/api/event/interaction/ApplicationCommandCreateEvent.java (hunk 1 above)
Comment (KILLEliteMaste) [reply javacord-api/src/main/java/org/javacord/api/event/interaction/ApplicationCommandCreateEvent.java:18]: But removing it from the new events result in the same. If someone want to pass the interaction to somewhere he might want to get the top level interaciton. The only difference is that the event is of this specific type but if the user wants to store it or has a global interacion handler he could get the top level interaction. So imo either remove it from all events or none

Comment (coaxial1) [reply javacord-api/src/main/java/org/javacord/api/event/interaction/ApplicationCommandCreateEvent.java:18]: Okay, agreed.
Then my only objection is that the type specfic interaction getter shouldn't return optional here, but the interaction directly, like you did in the other events :)

Comment (KILLEliteMaste) [reply javacord-api/src/main/java/org/javacord/api/event/interaction/ApplicationCommandCreateEvent.java:18]: Oh yeah, you are right. Missed that somehow :)

**Pull request merged**